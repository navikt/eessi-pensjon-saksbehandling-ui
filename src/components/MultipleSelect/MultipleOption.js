import React from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import { Checkbox } from 'components/Nav'

const MultipleOption = (props) => {
  const { data, selectProps, innerProps, isSelected, isFocused } = props

  return <div id={selectProps.id + '-' + data.value}>
    <div className={classNames('c-multipleOption', {
      selected: isSelected,
      focused: isFocused
    })} {...innerProps}>
      <Checkbox
        className='c-multipleOption__checkbox'
        label={data.label}
        onChange={() => {}} checked={isSelected}
      />
    </div>
  </div>
}

MultipleOption.propTypes = {
  data: PT.object,
  selectProps: PT.object,
  innerProps: PT.object,
  isSelected: PT.bool,
  isFocused: PT.bool
}

export default MultipleOption
