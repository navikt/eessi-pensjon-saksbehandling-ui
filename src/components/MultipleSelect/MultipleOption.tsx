import { themeKeys } from 'nav-styled-component-theme'
import PT from 'prop-types'
import React from 'react'
import { OptionProps } from 'react-select'
import { Checkbox } from 'nav-frontend-skjema'
import styled, { ThemeProvider } from 'styled-components'

export type MultipleOptionProps = OptionProps<any>

const OptionCheckbox = styled(Checkbox)`
   &:not(:disabled) {
     cursor: pointer
   }
   label {
      &:hover {
       color: ${({ theme }) => theme[themeKeys.INVERTED_FONT_COLOR]} !important;
     }
   }
`

const MultipleOption: React.FC<MultipleOptionProps> = (props: MultipleOptionProps): JSX.Element => {
  const { data, selectProps, innerProps, isSelected, getStyles } = props
  const id: string = selectProps.id + '-' + data.value
  const theme = selectProps.selectProps.theme

  return (
    <div
      {...innerProps}
      style={getStyles('option', props)}
      onClick={(e) => {
        e.preventDefault()
        e.stopPropagation()
        innerProps.onClick(e)
      }}
    >
      <ThemeProvider theme={theme}>
        <OptionCheckbox
          data-test-id={'c-multipleOption__checkbox-' + id}
          label={data.label}
          onChange={(e) => {
            e.preventDefault()
            e.stopPropagation()
          }}
          checked={isSelected}
        />
      </ThemeProvider>
    </div>
  )
}

MultipleOption.propTypes = {
  data: PT.object,
  selectProps: PT.object.isRequired,
  innerProps: PT.any.isRequired,
  isSelected: PT.bool.isRequired,
  isFocused: PT.bool.isRequired
}
MultipleOption.displayName = 'MultipleOption'
export default MultipleOption
