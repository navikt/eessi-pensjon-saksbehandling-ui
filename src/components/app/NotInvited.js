import React, { Component } from 'react'
import { connect } from 'react-redux'
import PT from 'prop-types'
import { withNamespaces } from 'react-i18next'
import classNames from 'classnames'

import TopContainer from '../../components/ui/TopContainer/TopContainer'
import FrontPageDrawer from '../../components/drawer/FrontPage'

const mapStateToProps = (state) => {
  return {
    status: state.status
  }
}

class NotInvited extends Component {
  render () {
    const { t, history, location, status } = this.props

    return <TopContainer className={classNames('p-app-notInvited')}
      history={history} location={location}
      sideContent={<FrontPageDrawer t={t} status={status} />}>
      <h1 className='typo-sidetittel ml-0 appTitle'>{'not invited'}</h1>
    </TopContainer>
  }
}

NotInvited.propTypes = {
  t: PT.func.isRequired,
  history: PT.object,
  location: PT.object,
  status: PT.object
}

export default connect(
  mapStateToProps
)(
  withNamespaces()(NotInvited)
)
