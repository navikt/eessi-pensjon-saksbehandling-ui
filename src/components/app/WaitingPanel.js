import React, { Component } from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import { withNamespaces } from 'react-i18next'
import * as Nav from '../ui/Nav'

class WaitingPanel extends Component {
  render () {
    const { t, message } = this.props

    return <div className={classNames('text-center')}>
      <Nav.NavFrontendSpinner />
      <p className='typo-normal'>{t(message)}</p>
    </div>
  }
}

WaitingPanel.propTypes = {
  message: PT.string.isRequired
}

export default withNamespaces(WaitingPanel)
