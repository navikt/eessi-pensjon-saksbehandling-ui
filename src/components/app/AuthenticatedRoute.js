import React, { Component } from 'react'
import PT from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Route, withRouter, Redirect } from 'react-router'
import _ from 'lodash'

import { IS_DEVELOPMENT } from '../../constants/environment'
import WaitingPanel from './WaitingPanel'

import * as routes from '../../constants/routes'
import * as appActions from '../../actions/app'
import * as statusActions from '../../actions/status'

const mapStateToProps = (state) => {
  return {
    userRole: state.app.userRole,
    userStatus: state.app.userStatus,
    loggedIn: state.app.loggedIn,
    allowed: state.app.allowed,
    isLoggingIn: state.loading.isLoggingIn,
    gettingUserInfo: state.loading.gettingUserInfo,
    rinaId: state.status.rinaId
  }
}

const mapDispatchToProps = (dispatch) => {
  return { actions: bindActionCreators(Object.assign({}, appActions, statusActions), dispatch) }
}

const paramAliases = {
  'rinaid': 'rinaId',
  'saksNr': 'saksId',
  'fnr': 'aktoerId'
}

class AuthenticatedRoute extends Component {
  state = {
    isReady: false
  }

  parseSearchParams () {
    const { actions, location } = this.props

    let params = new URLSearchParams(location.search)
    let newParams = {}
    params.forEach((value, key) => {
      const _key = paramAliases.hasOwnProperty(key) ? paramAliases[key] : key
      const _value = value || undefined
      if (_value !== this.state[_key]) {
        actions.setStatusParam(_key, _value)
        newParams[_key] = _value
      }
    })
    if (!_.isEmpty(newParams)) {
      this.setState(newParams)
    }
    return newParams
  }

  componentDidMount () {
    const { actions, userStatus } = this.props
    if (!userStatus) {
      actions.getUserInfo()
    } else {
      this.setState({
        isReady: true
      })
    }
    this.parseSearchParams()
  }

  componentDidUpdate () {
    const { userStatus } = this.props
    const { isReady } = this.state

    if (!isReady && userStatus !== undefined) {
      this.setState({
        isReady: true
      })
    }
    this.parseSearchParams()
  }

  hasApprovedRole () {
    const { roles, userRole } = this.props
    return roles.indexOf(userRole) >= 0
  }

  comesFromPesys () {
    return this.state.hasOwnProperty('saksId') && this.state.hasOwnProperty('aktoerId')
  }

  forceLogin () {
    const { actions } = this.props

    console.log("No oidc-token, force login.")
    actions.login()
  }

  render () {
    const { userRole, allowed, gettingUserInfo } = this.props
    const { isReady } = this.state

    if (!isReady || gettingUserInfo) {
      return <WaitingPanel message='authenticating' />
    }

    if (!userRole) {
      this.forceLogin()
      return null
    }

    let validRole = this.hasApprovedRole()

    if (!validRole) {
      return <Redirect to={routes.FORBIDDEN} />
    }

    let authorized = allowed || IS_DEVELOPMENT

    if (!authorized) {
      return <Redirect to={routes.NOT_INVITED} />
    }

    return <Route {...this.props} />
  }
}

AuthenticatedRoute.propTypes = {
  className: PT.string,
  actions: PT.object.isRequired,
  roles: PT.array.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(
  withRouter(AuthenticatedRoute)
)
