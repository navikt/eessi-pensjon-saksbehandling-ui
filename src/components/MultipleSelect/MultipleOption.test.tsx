import { Option } from 'src/declarations/app'
import { render, screen } from '@testing-library/react'
import MultipleOption, { MultipleOptionProps } from './MultipleOption'

describe('components/MultipleSelect/MultipleOption', () => {
  const initialMockProps: MultipleOptionProps<Option> = {
    innerRef (): void {},
    // @ts-ignore
    selectProps: {},
    // @ts-ignore
    theme: undefined,
    id: undefined,
    isRtl: false,
    clearValue: jest.fn(),
    getValue: jest.fn(),
    setValue: jest.fn(),
    hasValue: false,
    selectOption: jest.fn(),
    options: [],
    isMulti: true,
    cx: jest.fn(),
    children: undefined,
    label: '',
    type: 'option',
    data: {
      label: 'mockLabel',
      value: 'mockValue'
    },
    getStyles: jest.fn(),
    innerProps: {
      onClick: jest.fn()
    } as any,
    isSelected: false,
    isFocused: false,
    isDisabled: false
  }

  it('Render: match snapshot', () => {
    const { container } = render(<MultipleOption {...initialMockProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    render(<MultipleOption {...initialMockProps} />)
    expect(screen.getByTestId('c-multipleoption--checkbox--mockValue')).toBeInTheDocument()
    screen.debug()
  })

  it('Handling: Triggers innerProps.onClick', () => {
    render(<MultipleOption {...initialMockProps} />)
    screen.getByTestId('c-multipleoption--checkbox--mockValue').click()
    expect(initialMockProps.innerProps.onClick).toHaveBeenCalled()
  })
})
