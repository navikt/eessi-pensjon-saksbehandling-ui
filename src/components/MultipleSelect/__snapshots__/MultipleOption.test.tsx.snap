// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MultipleSelect/MultipleOption Render: match snapshot 1`] = `
.c1:not(:disabled) {
  cursor: pointer;
}

.c1 label:hover {
  color: #FFFFFF !important;
}

ui-fss.g28[id="c0"] {
  content: "c1,";
}

<MultipleOption
  clearValue={[MockFunction]}
  cx={[MockFunction]}
  data={
    Object {
      "label": "mockLabel",
      "value": "mockValue",
    }
  }
  getStyles={
    [MockFunction] {
      "calls": Array [
        Array [
          "option",
          Object {
            "children": undefined,
            "clearValue": [MockFunction],
            "cx": [MockFunction],
            "data": Object {
              "label": "mockLabel",
              "value": "mockValue",
            },
            "getStyles": [MockFunction] {
              "calls": [Circular],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
            "getValue": [MockFunction],
            "hasValue": false,
            "innerProps": Object {
              "onClick": [MockFunction],
            },
            "innerRef": null,
            "isDisabled": false,
            "isFocused": false,
            "isMulti": true,
            "isSelected": false,
            "label": "",
            "options": Array [],
            "selectOption": [MockFunction],
            "selectProps": Object {
              "id": "mockId",
              "selectProps": Object {
                "theme": Object {
                  "alternative-background-color": "#e9e7e7",
                  "black": "#000000",
                  "border-radius-base": "4px",
                  "border-radius-large": "6px",
                  "border-radius-small": "2px",
                  "container-desktop": "(940px + @grid-gutter-width)",
                  "container-large-desktop": "(1140px + @grid-gutter-width)",
                  "container-lg": "(1140px + @grid-gutter-width)",
                  "container-md": "(940px + @grid-gutter-width)",
                  "container-sm": "(720px + @grid-gutter-width)",
                  "container-tablet": "(720px + @grid-gutter-width)",
                  "fokusFarge": "#01b19d",
                  "grayBackground": "#e9e7e7",
                  "grayIcon": "#b7b1a9",
                  "grayInactive": "#78706a",
                  "grayModia": "#333333",
                  "grid-columns": "12",
                  "grid-float-breakpoint": "768px",
                  "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
                  "grid-gutter-width": "16px",
                  "inverted-font-color": "#FFFFFF",
                  "line-height-large": "1.3333333",
                  "line-height-small": "1.5",
                  "main-active-color": "#43dfNaN",
                  "main-background-color": "#FFFFFF",
                  "main-border-color": "#78706a",
                  "main-border-radius": "4px",
                  "main-disabled-color": "#b7b1a9",
                  "main-error-color": "#ba3a26",
                  "main-focus-color": "#01b19d",
                  "main-font-color": "#3e3832",
                  "main-hover-color": "#20f4NaN",
                  "main-interactive-color": "#0067c5",
                  "main-success-color": "#06893a",
                  "main-warning-color": "#ff9100",
                  "navBakgrunn": "#ffffff",
                  "navBla": "#0067c5",
                  "navBlaDarken20": "#007fb7",
                  "navBlaDarken40": "#0198aa",
                  "navBlaDarken60": "#01b19d",
                  "navBlaDarken80": "#02ca90",
                  "navBlaLighten20": "#0385NaN",
                  "navBlaLighten40": "#06a3NaN",
                  "navBlaLighten60": "#09c2NaN",
                  "navBlaLighten80": "#0ce0NaN",
                  "navDypBla": "#005b82",
                  "navDypBlaDarken20": "#007682",
                  "navDypBlaDarken40": "#019182",
                  "navDypBlaDarken60": "#01ac82",
                  "navDypBlaDarken80": "#02c782",
                  "navDypBlaLighten20": "#037bNaN",
                  "navDypBlaLighten40": "#069cNaN",
                  "navDypBlaLighten60": "#09bdNaN",
                  "navDypBlaLighten80": "#0cdeNaN",
                  "navGra20": "#c6c2bf",
                  "navGra40": "#b7b1a9",
                  "navGra60": "#78706a",
                  "navGra80": "#59514b",
                  "navGraBakgrunn": "#f1f1f1",
                  "navGronn": "#06893a",
                  "navGronnDarken20": "#059b48",
                  "navGronnDarken40": "#04ad57",
                  "navGronnDarken60": "#04bf65",
                  "navGronnDarken80": "#03d174",
                  "navGronnLighten20": "#07a0NaN",
                  "navGronnLighten40": "#09b8NaN",
                  "navGronnLighten60": "#0bcfNaN",
                  "navGronnLighten80": "#0de7NaN",
                  "navLilla": "#634689",
                  "navLillaDarken20": "#4f6587",
                  "navLillaDarken40": "#3c8486",
                  "navLillaDarken60": "#29a485",
                  "navLillaDarken80": "#16c384",
                  "navLillaLighten20": "#526bNaN",
                  "navLillaLighten40": "#4190NaN",
                  "navLillaLighten60": "#30b5NaN",
                  "navLillaLighten80": "#1fdaNaN",
                  "navLimeGronn": "#a2ad00",
                  "navLimeGronnDarken20": "#82b71a",
                  "navLimeGronnDarken40": "#62c234",
                  "navLimeGronnDarken60": "#42cd4e",
                  "navLimeGronnDarken80": "#22d868",
                  "navLimeGronnLighten20": "#84bdNaN",
                  "navLimeGronnLighten40": "#67cdNaN",
                  "navLimeGronnLighten60": "#49deNaN",
                  "navLimeGronnLighten80": "#2ceeNaN",
                  "navLysBla": "#66cbec",
                  "navLysBlaDarken20": "#52cfd7",
                  "navLysBlaDarken40": "#3ed4c2",
                  "navLysBlaDarken60": "#2ad9ad",
                  "navLysBlaDarken80": "#16de98",
                  "navLysBlaLighten20": "#54d5NaN",
                  "navLysBlaLighten40": "#43dfNaN",
                  "navLysBlaLighten60": "#31eaNaN",
                  "navLysBlaLighten80": "#20f4NaN",
                  "navLysGra": "#e9e7e7",
                  "navMorkGra": "#3e3832",
                  "navOransje": "#ff9100",
                  "navOransjeDarken20": "#cca11a",
                  "navOransjeDarken40": "#9ab134",
                  "navOransjeDarken60": "#67c24e",
                  "navOransjeDarken80": "#35d268",
                  "navOransjeLighten20": "#cfa7NaN",
                  "navOransjeLighten40": "#9fbdNaN",
                  "navOransjeLighten60": "#6fd3NaN",
                  "navOransjeLighten80": "#3fe9NaN",
                  "navRod": "#c30000",
                  "navRodDarken20": "#9c2d1a",
                  "navRodDarken40": "#765a34",
                  "navRodDarken60": "#4f884e",
                  "navRodDarken80": "#29b568",
                  "navRodLighten20": "#9f33NaN",
                  "navRodLighten40": "#7b66NaN",
                  "navRodLighten60": "#5799NaN",
                  "navRodLighten80": "#33ccNaN",
                  "orangeFocus": "#ffbd66",
                  "orangeFocusDarken20": "#ccc46b",
                  "orangeFocusDarken40": "#9acc71",
                  "orangeFocusDarken60": "#67d377",
                  "orangeFocusDarken80": "#35db7d",
                  "orangeFocusLighten20": "#cfcaNaN",
                  "orangeFocusLighten40": "#9fd7NaN",
                  "orangeFocusLighten60": "#6fe4NaN",
                  "orangeFocusLighten80": "#3ff1NaN",
                  "padding-base-horizontal": "12px",
                  "padding-base-vertical": "6px",
                  "padding-large-horizontal": "16px",
                  "padding-large-vertical": "10px",
                  "padding-small-horizontal": "10px",
                  "padding-small-vertical": "5px",
                  "padding-xs-horizontal": "5px",
                  "padding-xs-vertical": "1px",
                  "pinkErrorBg": "#f3e3e3",
                  "redError": "#ba3a26",
                  "redErrorDarken20": "#955b38",
                  "redErrorDarken40": "#707d4b",
                  "redErrorDarken60": "#4c9f5d",
                  "redErrorDarken80": "#27c170",
                  "redErrorLighten20": "#9761NaN",
                  "redErrorLighten40": "#7588NaN",
                  "redErrorLighten60": "#53b0NaN",
                  "redErrorLighten80": "#31d7NaN",
                  "screen-desktop": "992px",
                  "screen-lg": "1200px",
                  "screen-lg-desktop": "1200px",
                  "screen-lg-min": "1200px",
                  "screen-md": "992px",
                  "screen-md-max": "(@screen-lg-min - 1)",
                  "screen-md-min": "992px",
                  "screen-sm": "768px",
                  "screen-sm-max": "(@screen-md-min - 1)",
                  "screen-sm-min": "768px",
                  "screen-tablet": "768px",
                  "screen-xs-max": "(@screen-sm-min - 1)",
                  "type": "theme",
                  "white": "#FFFFFF",
                },
              },
            },
            "setValue": [MockFunction],
            "type": "option",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  getValue={[MockFunction]}
  hasValue={false}
  innerProps={
    Object {
      "onClick": [MockFunction],
    }
  }
  innerRef={null}
  isDisabled={false}
  isFocused={false}
  isMulti={true}
  isSelected={false}
  label=""
  options={Array []}
  selectOption={[MockFunction]}
  selectProps={
    Object {
      "id": "mockId",
      "selectProps": Object {
        "theme": Object {
          "alternative-background-color": "#e9e7e7",
          "black": "#000000",
          "border-radius-base": "4px",
          "border-radius-large": "6px",
          "border-radius-small": "2px",
          "container-desktop": "(940px + @grid-gutter-width)",
          "container-large-desktop": "(1140px + @grid-gutter-width)",
          "container-lg": "(1140px + @grid-gutter-width)",
          "container-md": "(940px + @grid-gutter-width)",
          "container-sm": "(720px + @grid-gutter-width)",
          "container-tablet": "(720px + @grid-gutter-width)",
          "fokusFarge": "#01b19d",
          "grayBackground": "#e9e7e7",
          "grayIcon": "#b7b1a9",
          "grayInactive": "#78706a",
          "grayModia": "#333333",
          "grid-columns": "12",
          "grid-float-breakpoint": "768px",
          "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
          "grid-gutter-width": "16px",
          "inverted-font-color": "#FFFFFF",
          "line-height-large": "1.3333333",
          "line-height-small": "1.5",
          "main-active-color": "#43dfNaN",
          "main-background-color": "#FFFFFF",
          "main-border-color": "#78706a",
          "main-border-radius": "4px",
          "main-disabled-color": "#b7b1a9",
          "main-error-color": "#ba3a26",
          "main-focus-color": "#01b19d",
          "main-font-color": "#3e3832",
          "main-hover-color": "#20f4NaN",
          "main-interactive-color": "#0067c5",
          "main-success-color": "#06893a",
          "main-warning-color": "#ff9100",
          "navBakgrunn": "#ffffff",
          "navBla": "#0067c5",
          "navBlaDarken20": "#007fb7",
          "navBlaDarken40": "#0198aa",
          "navBlaDarken60": "#01b19d",
          "navBlaDarken80": "#02ca90",
          "navBlaLighten20": "#0385NaN",
          "navBlaLighten40": "#06a3NaN",
          "navBlaLighten60": "#09c2NaN",
          "navBlaLighten80": "#0ce0NaN",
          "navDypBla": "#005b82",
          "navDypBlaDarken20": "#007682",
          "navDypBlaDarken40": "#019182",
          "navDypBlaDarken60": "#01ac82",
          "navDypBlaDarken80": "#02c782",
          "navDypBlaLighten20": "#037bNaN",
          "navDypBlaLighten40": "#069cNaN",
          "navDypBlaLighten60": "#09bdNaN",
          "navDypBlaLighten80": "#0cdeNaN",
          "navGra20": "#c6c2bf",
          "navGra40": "#b7b1a9",
          "navGra60": "#78706a",
          "navGra80": "#59514b",
          "navGraBakgrunn": "#f1f1f1",
          "navGronn": "#06893a",
          "navGronnDarken20": "#059b48",
          "navGronnDarken40": "#04ad57",
          "navGronnDarken60": "#04bf65",
          "navGronnDarken80": "#03d174",
          "navGronnLighten20": "#07a0NaN",
          "navGronnLighten40": "#09b8NaN",
          "navGronnLighten60": "#0bcfNaN",
          "navGronnLighten80": "#0de7NaN",
          "navLilla": "#634689",
          "navLillaDarken20": "#4f6587",
          "navLillaDarken40": "#3c8486",
          "navLillaDarken60": "#29a485",
          "navLillaDarken80": "#16c384",
          "navLillaLighten20": "#526bNaN",
          "navLillaLighten40": "#4190NaN",
          "navLillaLighten60": "#30b5NaN",
          "navLillaLighten80": "#1fdaNaN",
          "navLimeGronn": "#a2ad00",
          "navLimeGronnDarken20": "#82b71a",
          "navLimeGronnDarken40": "#62c234",
          "navLimeGronnDarken60": "#42cd4e",
          "navLimeGronnDarken80": "#22d868",
          "navLimeGronnLighten20": "#84bdNaN",
          "navLimeGronnLighten40": "#67cdNaN",
          "navLimeGronnLighten60": "#49deNaN",
          "navLimeGronnLighten80": "#2ceeNaN",
          "navLysBla": "#66cbec",
          "navLysBlaDarken20": "#52cfd7",
          "navLysBlaDarken40": "#3ed4c2",
          "navLysBlaDarken60": "#2ad9ad",
          "navLysBlaDarken80": "#16de98",
          "navLysBlaLighten20": "#54d5NaN",
          "navLysBlaLighten40": "#43dfNaN",
          "navLysBlaLighten60": "#31eaNaN",
          "navLysBlaLighten80": "#20f4NaN",
          "navLysGra": "#e9e7e7",
          "navMorkGra": "#3e3832",
          "navOransje": "#ff9100",
          "navOransjeDarken20": "#cca11a",
          "navOransjeDarken40": "#9ab134",
          "navOransjeDarken60": "#67c24e",
          "navOransjeDarken80": "#35d268",
          "navOransjeLighten20": "#cfa7NaN",
          "navOransjeLighten40": "#9fbdNaN",
          "navOransjeLighten60": "#6fd3NaN",
          "navOransjeLighten80": "#3fe9NaN",
          "navRod": "#c30000",
          "navRodDarken20": "#9c2d1a",
          "navRodDarken40": "#765a34",
          "navRodDarken60": "#4f884e",
          "navRodDarken80": "#29b568",
          "navRodLighten20": "#9f33NaN",
          "navRodLighten40": "#7b66NaN",
          "navRodLighten60": "#5799NaN",
          "navRodLighten80": "#33ccNaN",
          "orangeFocus": "#ffbd66",
          "orangeFocusDarken20": "#ccc46b",
          "orangeFocusDarken40": "#9acc71",
          "orangeFocusDarken60": "#67d377",
          "orangeFocusDarken80": "#35db7d",
          "orangeFocusLighten20": "#cfcaNaN",
          "orangeFocusLighten40": "#9fd7NaN",
          "orangeFocusLighten60": "#6fe4NaN",
          "orangeFocusLighten80": "#3ff1NaN",
          "padding-base-horizontal": "12px",
          "padding-base-vertical": "6px",
          "padding-large-horizontal": "16px",
          "padding-large-vertical": "10px",
          "padding-small-horizontal": "10px",
          "padding-small-vertical": "5px",
          "padding-xs-horizontal": "5px",
          "padding-xs-vertical": "1px",
          "pinkErrorBg": "#f3e3e3",
          "redError": "#ba3a26",
          "redErrorDarken20": "#955b38",
          "redErrorDarken40": "#707d4b",
          "redErrorDarken60": "#4c9f5d",
          "redErrorDarken80": "#27c170",
          "redErrorLighten20": "#9761NaN",
          "redErrorLighten40": "#7588NaN",
          "redErrorLighten60": "#53b0NaN",
          "redErrorLighten80": "#31d7NaN",
          "screen-desktop": "992px",
          "screen-lg": "1200px",
          "screen-lg-desktop": "1200px",
          "screen-lg-min": "1200px",
          "screen-md": "992px",
          "screen-md-max": "(@screen-lg-min - 1)",
          "screen-md-min": "992px",
          "screen-sm": "768px",
          "screen-sm-max": "(@screen-md-min - 1)",
          "screen-sm-min": "768px",
          "screen-tablet": "768px",
          "screen-xs-max": "(@screen-sm-min - 1)",
          "type": "theme",
          "white": "#FFFFFF",
        },
      },
    }
  }
  setValue={[MockFunction]}
  type="option"
>
  <div
    data-test-id="c-multipleoption__div-id"
    onClick={[Function]}
    role="button"
  >
    <Xe>
      <Component
        theme={
          Object {
            "alternative-background-color": "#e9e7e7",
            "black": "#000000",
            "border-radius-base": "4px",
            "border-radius-large": "6px",
            "border-radius-small": "2px",
            "container-desktop": "(940px + @grid-gutter-width)",
            "container-large-desktop": "(1140px + @grid-gutter-width)",
            "container-lg": "(1140px + @grid-gutter-width)",
            "container-md": "(940px + @grid-gutter-width)",
            "container-sm": "(720px + @grid-gutter-width)",
            "container-tablet": "(720px + @grid-gutter-width)",
            "fokusFarge": "#01b19d",
            "grayBackground": "#e9e7e7",
            "grayIcon": "#b7b1a9",
            "grayInactive": "#78706a",
            "grayModia": "#333333",
            "grid-columns": "12",
            "grid-float-breakpoint": "768px",
            "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
            "grid-gutter-width": "16px",
            "inverted-font-color": "#FFFFFF",
            "line-height-large": "1.3333333",
            "line-height-small": "1.5",
            "main-active-color": "#43dfNaN",
            "main-background-color": "#FFFFFF",
            "main-border-color": "#78706a",
            "main-border-radius": "4px",
            "main-disabled-color": "#b7b1a9",
            "main-error-color": "#ba3a26",
            "main-focus-color": "#01b19d",
            "main-font-color": "#3e3832",
            "main-hover-color": "#20f4NaN",
            "main-interactive-color": "#0067c5",
            "main-success-color": "#06893a",
            "main-warning-color": "#ff9100",
            "navBakgrunn": "#ffffff",
            "navBla": "#0067c5",
            "navBlaDarken20": "#007fb7",
            "navBlaDarken40": "#0198aa",
            "navBlaDarken60": "#01b19d",
            "navBlaDarken80": "#02ca90",
            "navBlaLighten20": "#0385NaN",
            "navBlaLighten40": "#06a3NaN",
            "navBlaLighten60": "#09c2NaN",
            "navBlaLighten80": "#0ce0NaN",
            "navDypBla": "#005b82",
            "navDypBlaDarken20": "#007682",
            "navDypBlaDarken40": "#019182",
            "navDypBlaDarken60": "#01ac82",
            "navDypBlaDarken80": "#02c782",
            "navDypBlaLighten20": "#037bNaN",
            "navDypBlaLighten40": "#069cNaN",
            "navDypBlaLighten60": "#09bdNaN",
            "navDypBlaLighten80": "#0cdeNaN",
            "navGra20": "#c6c2bf",
            "navGra40": "#b7b1a9",
            "navGra60": "#78706a",
            "navGra80": "#59514b",
            "navGraBakgrunn": "#f1f1f1",
            "navGronn": "#06893a",
            "navGronnDarken20": "#059b48",
            "navGronnDarken40": "#04ad57",
            "navGronnDarken60": "#04bf65",
            "navGronnDarken80": "#03d174",
            "navGronnLighten20": "#07a0NaN",
            "navGronnLighten40": "#09b8NaN",
            "navGronnLighten60": "#0bcfNaN",
            "navGronnLighten80": "#0de7NaN",
            "navLilla": "#634689",
            "navLillaDarken20": "#4f6587",
            "navLillaDarken40": "#3c8486",
            "navLillaDarken60": "#29a485",
            "navLillaDarken80": "#16c384",
            "navLillaLighten20": "#526bNaN",
            "navLillaLighten40": "#4190NaN",
            "navLillaLighten60": "#30b5NaN",
            "navLillaLighten80": "#1fdaNaN",
            "navLimeGronn": "#a2ad00",
            "navLimeGronnDarken20": "#82b71a",
            "navLimeGronnDarken40": "#62c234",
            "navLimeGronnDarken60": "#42cd4e",
            "navLimeGronnDarken80": "#22d868",
            "navLimeGronnLighten20": "#84bdNaN",
            "navLimeGronnLighten40": "#67cdNaN",
            "navLimeGronnLighten60": "#49deNaN",
            "navLimeGronnLighten80": "#2ceeNaN",
            "navLysBla": "#66cbec",
            "navLysBlaDarken20": "#52cfd7",
            "navLysBlaDarken40": "#3ed4c2",
            "navLysBlaDarken60": "#2ad9ad",
            "navLysBlaDarken80": "#16de98",
            "navLysBlaLighten20": "#54d5NaN",
            "navLysBlaLighten40": "#43dfNaN",
            "navLysBlaLighten60": "#31eaNaN",
            "navLysBlaLighten80": "#20f4NaN",
            "navLysGra": "#e9e7e7",
            "navMorkGra": "#3e3832",
            "navOransje": "#ff9100",
            "navOransjeDarken20": "#cca11a",
            "navOransjeDarken40": "#9ab134",
            "navOransjeDarken60": "#67c24e",
            "navOransjeDarken80": "#35d268",
            "navOransjeLighten20": "#cfa7NaN",
            "navOransjeLighten40": "#9fbdNaN",
            "navOransjeLighten60": "#6fd3NaN",
            "navOransjeLighten80": "#3fe9NaN",
            "navRod": "#c30000",
            "navRodDarken20": "#9c2d1a",
            "navRodDarken40": "#765a34",
            "navRodDarken60": "#4f884e",
            "navRodDarken80": "#29b568",
            "navRodLighten20": "#9f33NaN",
            "navRodLighten40": "#7b66NaN",
            "navRodLighten60": "#5799NaN",
            "navRodLighten80": "#33ccNaN",
            "orangeFocus": "#ffbd66",
            "orangeFocusDarken20": "#ccc46b",
            "orangeFocusDarken40": "#9acc71",
            "orangeFocusDarken60": "#67d377",
            "orangeFocusDarken80": "#35db7d",
            "orangeFocusLighten20": "#cfcaNaN",
            "orangeFocusLighten40": "#9fd7NaN",
            "orangeFocusLighten60": "#6fe4NaN",
            "orangeFocusLighten80": "#3ff1NaN",
            "padding-base-horizontal": "12px",
            "padding-base-vertical": "6px",
            "padding-large-horizontal": "16px",
            "padding-large-vertical": "10px",
            "padding-small-horizontal": "10px",
            "padding-small-vertical": "5px",
            "padding-xs-horizontal": "5px",
            "padding-xs-vertical": "1px",
            "pinkErrorBg": "#f3e3e3",
            "redError": "#ba3a26",
            "redErrorDarken20": "#955b38",
            "redErrorDarken40": "#707d4b",
            "redErrorDarken60": "#4c9f5d",
            "redErrorDarken80": "#27c170",
            "redErrorLighten20": "#9761NaN",
            "redErrorLighten40": "#7588NaN",
            "redErrorLighten60": "#53b0NaN",
            "redErrorLighten80": "#31d7NaN",
            "screen-desktop": "992px",
            "screen-lg": "1200px",
            "screen-lg-desktop": "1200px",
            "screen-lg-min": "1200px",
            "screen-md": "992px",
            "screen-md-max": "(@screen-lg-min - 1)",
            "screen-md-min": "992px",
            "screen-sm": "768px",
            "screen-sm-max": "(@screen-md-min - 1)",
            "screen-sm-min": "768px",
            "screen-tablet": "768px",
            "screen-xs-max": "(@screen-sm-min - 1)",
            "type": "theme",
            "white": "#FFFFFF",
          }
        }
      >
        <Styled(Checkbox)
          checked={false}
          data-test-id="c-multipleoption__checkbox-mockId-mockValue"
          id="c-multipleoption__checkbox-mockId-mockValue"
          label="mockLabel"
          onChange={[Function]}
        >
          <Checkbox
            checked={false}
            className="c0 c1"
            data-test-id="c-multipleoption__checkbox-mockId-mockValue"
            id="c-multipleoption__checkbox-mockId-mockValue"
            label="mockLabel"
            onChange={[Function]}
          >
            <div
              className="skjemaelement c0 c1"
            >
              <input
                aria-invalid={false}
                checked={false}
                className="skjemaelement__input checkboks"
                data-test-id="c-multipleoption__checkbox-mockId-mockValue"
                id="c-multipleoption__checkbox-mockId-mockValue"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="skjemaelement__label"
                htmlFor="c-multipleoption__checkbox-mockId-mockValue"
              >
                mockLabel
              </label>
            </div>
          </Checkbox>
        </Styled(Checkbox)>
      </Component>
    </Xe>
  </div>
</MultipleOption>
`;
